apps/explorer/lib/explorer/market/market.ex

```diff
@@ -40,4 +53,32 @@ defmodule Explorer.Market do
   def bulk_insert_history(records) do
     Repo.insert_all(MarketHistory, records, on_conflict: :replace_all, conflict_target: [:date])
   end
+
+  def add_price(%{symbol: symbol} = token) do
+    known_address = get_known_address(symbol)
+
+    matches_known_address = known_address && known_address == token.contract_address_hash
+
+    usd_value =
+      if matches_known_address do
```

Comment:
🎨 you can avoid this `if` by using `pattern matching`

```suggestion
  usd_value = fetch_token_usd_value(matches_known_address, symbol)
  
  defp fetch_token_usd_value(true, symbol) do
    case get_exchange_rate(symbol) do
      %{usd_value: usd_value} -> usd_value
      nil -> nil
    end
  end

  defp fetch_token_usd_value(_matches_known_address, symbol), do: nil
```

Reply:
Done, thanks!